libbeat/outputs/elasticsearch/elasticsearch.go

```diff
@@ -50,19 +50,40 @@ var (
 )
 
 type callbacksRegistry struct {
-	callbacks []connectCallback
+	callbacks map[int]connectCallback
```

Comment:
What about using a uuid here instead of an int as identifier? Like this the incrementing and storing of the current key would not be needed.